src/components/DryRun/DryRun.tsx

```diff
@@ -0,0 +1,169 @@
+import Grid from "@dashboard/components/Grid";
+import { useStyles } from "@dashboard/custom-apps/components/WebhookEvents/styles";
+import {
+  useTriggerWebhookDryRunMutation,
+  WebhookEventTypeAsyncEnum,
+} from "@dashboard/graphql";
+import {
+  capitalize,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  Typography,
+} from "@material-ui/core";
+import {
+  Alert,
+  Button,
+  DialogHeader,
+  List,
+  ListBody,
+  ListHeader,
+  ListItem,
+  ListItemCell,
+} from "@saleor/macaw-ui";
+import React, { useState } from "react";
+import { useIntl } from "react-intl";
+
+import DryRunItemsList from "../DryRunItemsList/DryRunItemsList";
+import { messages } from "./messages";
+import { getObjects } from "./utils";
+
+interface DryRunProps {
+  query: string;
+  showDialog: boolean;
+  setShowDialog: React.Dispatch<React.SetStateAction<boolean>>;
+  asyncEvents: WebhookEventTypeAsyncEnum[];
+  setResult: React.Dispatch<React.SetStateAction<string>>;
+}
+
+const DryRun = (props: DryRunProps) => {
+  const intl = useIntl();
+  const { query, showDialog, setShowDialog } = props;
+  const classes = useStyles();
+  const [objectId, setObjectId] = useState(null);
+  const [triggerWebhookDryRun] = useTriggerWebhookDryRunMutation();
+  const availableObjects = getObjects(query);
+  const unavailableObjects = getObjects(query, false);
+
+  const [object, setObject] = useState(null);
+
+  const dryRun = async () => {
+    const { data } = await triggerWebhookDryRun({
+      variables: { objectId, query },
+    });
+    props.setResult(
+      JSON.stringify(JSON.parse(data.webhookDryRun.payload), null, 2),
+    );
+    closeDialog();
+  };
+
+  const closeDialog = () => {
+    setShowDialog(false);
+    setObjectId(null);
+    setObject(null);
+    props.setShowDialog(false);
+  };
+
+  if (!showDialog) {
+    return <></>;
+  }
+
+  return (
+    <Dialog open={showDialog} fullWidth maxWidth="md" data-test-id="dry-run">
+      <DialogHeader onClose={closeDialog}>
+        {intl.formatMessage(messages.header)}
+      </DialogHeader>
+      <DialogContent style={{ overflow: "scroll" }}>
+        <DialogContentText>
+          {intl.formatMessage(messages.selectObject)}
+        </DialogContentText>
+
+        {unavailableObjects.length && (
```

Comment:
question: what in case that length will be 0, will not be displayed?

Reply:
Nice catch!